### DISCUSSION TITLE
GpsRequest delegate to frequently on iOS

### DISCUSSION THREAD

**Discussion by dxt-rr** [shinyorg/shiny#780] [Q&A] [answered]
Hello there,

I have a little problem with Shiny background GPS. Generally it is working very well, I do get the coordinates in background, but just to often. I'm using the provided sample with just one change creating the request:
`var request = new GpsRequest { UseBackground = true, Priority = GpsPriority.Normal, Interval = TimeSpan.FromSeconds (60), ThrottledInterval = TimeSpan.FromSeconds (30), MinimumDistance = Distance.FromMeters (5) };`

On Android it is working as expected. The delegate is fired in an interval not below 30 seconds and I have to move araund to make the delegate get called.

On iOS I start the listener, press the home button to move the app into background and I lock the screen. Even if the iPhone or iPad is laying on the desk the delegate will get fired every second. Only at night it slows down to minutes or hours. No matter the battery state or of I have started other apps. Only if the battery is extremly low, the app stops receiving after about 20 minutes.

Of course I can reduce the amount of stored data manually by checking the time stamp or calculate the distance manually using the GeoCoordinates class, but that would be code on my side and waste battery...
Am I doing something wrong or did I miss any specifications? Is it, because I've deployed in debug mode (of course I disconnect the cable before testing)?

Does anybody have any clue?

Thank's
Ralph

**Comment by aritchie** [comment, marked as answer]
There was a couple of fixes in 2.4 which was released yesterday that should also fix this.

**Reply by dxt-rr** [reply]
Works as expected now with 2.4! Thank you very much for your effort.
